### ISSUE TITLE
New command: Eval wrapped

### ISSUE THREAD

**Issue by tonsky** [tonsky/Clojure-Sublimed#101]
The idea is you select some code, execute this command, and it does something like `eval (println %)` where % is selected code

Can use it to:

- print evaled code to console
- pretty-print evaled code to console
- maybe other use-cases?

**Comment by KGOH** [comment]
I would use this in conjuction with my debug macroses and profiling functions

I would like to be able to configure a separate eval keybinding with format string provided, like
```
 {"keys": ["ctrl+shift+enter"],
  "command": "clojure_sublimed_eval",
  "args": {"wrap_fstr": "(doto %s clojure.pprint/pprint)"}
  "context": [{"key": "selector", "operator": "equal", "operand": "source.clojure"}]},
```

Then if I execute this command on some code, like 
```
(+ 2 3)
```
I expect the plugin to wrap this into 
```
(doto (+ 2 3) clojure.pprint/pprint)
```
then execute and return result of this modified form

**Comment by KGOH** [comment]
I also propose to add `warp_fn` instead of `wrap_fstr` and move the fstr functionality to a separate command. Thus progammatic calls state.conn.eval would be possible with more complex transformations than simple format string wrap. For example, insert #p (https://github.com/weavejester/hashp) debug print

Here's the diff on the #102 
```diff
diff --git a/cs_conn.py b/cs_conn.py
index f1a8369..ec3d270 100644
--- a/cs_conn.py
+++ b/cs_conn.py
@@ -47,19 +47,19 @@ class Connection:
             return cs_parser.topmost_form(view, region.begin())
         return region
 
-    def eval(self, view, sel, wrap_fstr=None):
+    def eval(self, view, sel, wrap_fn=None):
         """
         Eval code and call `cs_eval.on_success(id, value)` or `cs_eval.on_exception(id, value, trace)`
         """
-        for region in sel:
-            region = self.eval_region(region, view)
+        for sel_region in sel:
+            region = self.eval_region(sel_region, view)
             eval = cs_eval.Eval(view, region)
             (line, column) = view.rowcol_utf16(region.begin())
             line = line + 1
 
             code = view.substr(region)
-            if wrap_fstr is not None:
-                code = wrap_fstr%code
+            if wrap_fn is not None:
+                code = wrap_fn(code, region, sel_region)
 
             form = cs_common.Form(
                     id     = eval.id,
diff --git a/cs_conn_socket_repl.py b/cs_conn_socket_repl.py
index 81ce120..388449d 100644
--- a/cs_conn_socket_repl.py
+++ b/cs_conn_socket_repl.py
@@ -81,11 +81,11 @@ class ConnectionSocketRepl(cs_conn.Connection):
         msg += '}'
         self.send(msg)
 
-    def eval(self, view, sel, wrap_fstr=None):
+    def eval(self, view, sel, wrap_fn=None):
         cs_warn.reset_warnings(self.window)
-        for region in sel:
+        for sel_region in sel:
             # find regions to eval
-            region = self.eval_region(region, view)
+            region = self.eval_region(sel_region, view)
 
             start = region.begin()
             parsed = cs_parser.parse(view.substr(region))
@@ -101,8 +101,8 @@ class ConnectionSocketRepl(cs_conn.Connection):
                 eval = cs_eval.Eval(view, form, id = f'{batch_id}.{idx}', batch_id = batch_id)
 
             code = view.substr(region)
-            if wrap_fstr is not None:
-                code = wrap_fstr%code
+            if wrap_fn is not None:
+                code = wrap_fn(code, region, sel_region)
             code = code.replace('\\', '\\\\').replace('"', '\\"')
 
             # send msg
diff --git a/cs_eval.py b/cs_eval.py
index c853cee..daac6a6 100644
--- a/cs_eval.py
+++ b/cs_eval.py
@@ -295,9 +295,20 @@ class ClojureSublimedEval(sublime_plugin.TextCommand):
     """
     Eval selected code or topmost form is selection is collapsed
     """
-    def run(self, edit, wrap_fstr=None):
+    def run(self, edit):
+        state = cs_common.get_state(self.view.window())
+        state.conn.eval(self.view, self.view.sel())
+
+    def is_enabled(self):
+        return cs_conn.ready(self.view.window())
+
+class ClojureSublimedEvalWithWrapCommand(sublime_plugin.TextCommand):
+    """
+    Wrap selected code or topmost form with format string and eval
+    """
+    def run(self, edit, wrap_fstr):
         state = cs_common.get_state(self.view.window())
-        state.conn.eval(self.view, self.view.sel(), wrap_fstr)
+        state.conn.eval(self.view, self.view.sel(), wrap_fn=(lambda code, _eval_region, _sel_region: wrap_fstr%code))
 
     def is_enabled(self):
         return cs_conn.ready(self.view.window())
```

This enabled me to define my custom command this way:
```python
def insert_code(code, eval_region, sel_region, insert_str):
    print(code, eval_region, sel_region, insert_str)
    global_cursor_pos = sel_region.begin()
    local_cursor_pos = global_cursor_pos - eval_region.begin()
    return code[:local_cursor_pos] + insert_str + code[local_cursor_pos:]


class MyClojureSublimedEvalWithInsertCommand(sublime_plugin.TextCommand):
    """ Evals with insert_str in the cursor position """
    def run(self, edit, insert_str):
        view = self.view
        state = cs_common.get_state(view.window())
        state.conn.eval(view, view.sel(), wrap_fn=(lambda code, eval_region, sel_region: insert_code(code, eval_region, sel_region, insert_str)))

    def is_enabled(self):
        return cs_conn.ready(self.view.window()) and len(self.view.sel()) == 1
```

Tested with raw nREPL and socket REPL connections

**Comment by tonsky** [comment]
Done in #102 and published in 3.6.0
